task3.py

```diff
@@ -0,0 +1,322 @@
+
+    Parameters:
+        csv_file (str): The path to the CSV file to be filtered
+        gender (str): The gender to filter by. If None, no gender filtering will be applied
+        rows (int): The maximum number of rows to include in the filtered data. If None, all rows will be included
+
+    Returns:
+        list: A list of dictionaries representing the filtered data
+    """
+    filtered_data = []
+    with open(csv_file, 'r', encoding='utf-8') as csv_file:
+        reader = csv.DictReader(csv_file)
+        for row in reader:
+            if (not gender or row['gender'].lower() == gender.lower()) and (not rows or reader.line_num <= rows):
+                filtered_data.append(row)
+
+    return filtered_data
+
+
+def add_fields(data):
+    """
+    Adds additional fields to the data
+
+    Parameters:
+        data (list): A list of dictionaries representing the data to which additional fields will be added
+    """
+    for row in data:
+        row['global_index'] = row.get('index', '')
+
+        dob_date = datetime.strptime(row['dob.date'], '%Y-%m-%dT%H:%M:%S.%fZ')
+        row['dob.date'] = dob_date.strftime('%m/%d/%Y')
+
+        reg_date = datetime.strptime(row['registered.date'], '%Y-%m-%dT%H:%M:%S.%fZ')
+        row['registered.date'] = reg_date.strftime('%m-%d-%Y, %H:%M:%S')
+
+        title_mapping = {
+            'Mrs': 'missis',
+            'Ms': 'miss',
+            'Mr': 'mister',
+            'Madame': 'mademoiselle'
+        }
+        row['name.title'] = title_mapping.get(row['name.title'])
+
+        current_time = datetime.utcnow().strftime('%Y-%m-%d %H:%M:%S')
+        row['current_time'] = current_time
+
+
+def create_folder_structure(data):
+    """
+    Creates folder structure based on data
+
+    Parameters:
+        data (list): A list of dictionaries representing the data used to create the folder structure
+
+    Returns:
+        dict: A nested dictionary representing the folder structure
+    """
+    folder_structure = {}
+    for row in data:
+        dob_year = datetime.strptime(row['dob.date'], '%m/%d/%Y').year
+        decade = f'{dob_year // 10}0-th'
+        country = row['location.country']
+
+        if decade not in folder_structure:
+            folder_structure[decade] = {}
+        if country not in folder_structure[decade]:
+            folder_structure[decade][country] = []
+
+        folder_structure[decade][country].append(row)
+
+    return folder_structure
+
+
+def calculate_max_age(data):
+    """
+    Calculates the maximum age from a list of data
+
+    Parameters:
+        data (list): A list of dictionaries representing the data containing date of birth information
+
+    Returns:
+        int: The maximum age calculated from the data
+    """
+    dob_dates = [datetime.strptime(row['dob.date'], '%m/%d/%Y') for row in data]
+    today = datetime.today()
+    max_age = max(today.year - date.year - ((today.month, today.day) < (date.month, date.day)) for date in dob_dates)
```

Comment:
I'm not sure, but I believe you can do: **date - date**. As a result U'll receive tamdelta obj and you could apply max on it.
in other words it could be simplified to one line